torch/testing/_internal/optests/generate_tests.py

```diff
@@ -146,6 +146,21 @@ def new_method(*args, **kwargs):
                 result = method(*args, **kwargs)
             return result
 
+        if pytestmark := new_method.__dict__.get("pytestmark"):
+            import pytest
+
+            new_pytestmark = []
+            for mark in pytestmark:
+                if isinstance(mark, pytest.Mark) and mark.name == "parametrize":
+                    argnames, argvalues = mark.args
+                    assert not mark.kwargs, "NYI"
+                    new_pytestmark.append(
+                        pytest.mark.parametrize(argnames, (next(iter(argvalues)),))
+                    )
+                else:
+                    new_pytestmark.append(mark)
+            new_method.__dict__["pytestmark"] = new_pytestmark
+
```

Comment:
I've thought about this before, especially because the internal tests also use some parametrization that make the tests go by slower. It kind of depends on how the parametrization is used:
- if it is just to generate different random values, then yes, one parametrization is sufficient.
- if it is to generate different options in your operator (e.g. BatchNorm with training=True/False), then it is important that we test all of them (BatchNorm mutates in one path but not the other)

Since generate_opcheck_tests is meant to be a "slow but runs everything check, because the user didn't want to write OpInfos (and we don't have an OpInfo V2 API)", I think I'd prefer the default to be "we run all parameterizations", but if the user understands what their test is doing and how it exercises their operator, they should be able to control the behavior to run just a single parametrization, perhaps by adding a decorator on the original test. WDYT?

Reply:
This is fine by me. I'll need to figure out a mechanism to add extra annotations to parametrize though.